core/src/main/scala/cats/syntax/ior.scala

```diff
@@ -0,0 +1,158 @@
+package cats.syntax
+
+import cats.data.{Ior, IorNel}
+
+trait IorSyntax {
+  implicit def catsSyntaxIorId[A](a: A): IorIdOps[A] = new IorIdOps(a)
+  implicit def catsSyntaxListIorNel[A, B](list: List[IorNel[A, B]]): IorNelListOps[A, B] =
+    new IorNelListOps(list)
+}
+
+final class IorIdOps[A](val a: A) extends AnyVal {
+  /**
+    * Wrap a value in `Ior.Right`.
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.Ior
+    * scala> import cats.implicits._
+    *
+    * scala> "hello".rightIor[String]
+    * res0: Ior[String, String] = Right(hello)
+    * }}}
+    */
+  def rightIor[B]: Ior[B, A] = Ior.right(a)
```

Comment:
Context in Either instances uses `asRight` so perhaps `asRightIor`. Depends on what sort of syntax maintainers would like to see but this makes more sense to me.